_posts/2017/2017-12-11-set-title-in-the-view-template-using-content-for.md

```diff
@@ -0,0 +1,71 @@
+---
+title: 'Set your page title in the view template using content_for'
+description: 'Cut out all that instance variable nonsense.'
+layout: article
+category: ruby
+image:
+  base: '2017/set-title-in-the-view-template-using-content-for'
+  alt: 'Books'
+  credit: 'Annie Spratt'
+  source: 'https://unsplash.com/photos/-9vMBjrU-RA'
+---
+
+Rails has `content_for` as it’s primary way to store content in one place for use in other views, layouts or helpers. See `content_for` in the Rails documentation at [in the ActionView helpers](http://api.rubyonrails.org/v5.1.3/classes/ActionView/Helpers/CaptureHelper.html#method-i-content_for).
+
+## Instead of…
+
+...using an instance variable in the controller.
+
+### `things_controller.rb`
+
+```ruby
+def show
+  @page_title = 'Page Title'
+  # other controller stuff
+end
+```
+
+### And `views/layouts/application.html.erb`
+
+In the `<head>`.
+
+```erb
+<title><%= @page_title || 'Default' %></title>
+```
+
+
+## Use…
+
+...`content_for` and `yield`.
+
+### `things/show.html.erb`
+
+I like to put the title at the top of the individual view file.
+
+```erb
+<% content_for(:html_title) { 'Title' } %>
+```
+
+### And `views/layouts/application.html.erb`
+
+In the `<head>`.
+
+```erb
+<title><%= content_for?(:html_title) ? yield(:page_title) : "Default" %></title>
+```
+
+
+## But why?
+
+Using `content_for` in this way keeps information about how to render the page inside the relevant view template. It also avoids cluttering the controller with non-business logic.
```

Comment:
Take out 'also' in the second sentence here, i.e. "It avoids cluttering..." This is because I think this is one main point, i.e: this is the right place for it to go. Or you can make it one long sentence: "...inside the relevant view template and it avoids cluttering the controller..." Up to you!

You also have an 'also' in the next paragraph..so taking it out here helps it to flow better.